ros2_foxglove_bridge/src/message_definition_cache.cpp

```diff
@@ -177,14 +185,19 @@ const MessageSpec& MessageDefinitionCache::load_message_spec(
   const std::string package = match[1].str();
   const std::string subfolder = match[2].str();
   const std::string type_name = match[3].str();
-  const bool is_action_type = subfolder == "action";
 
-  // The action type name includes the subtype which we have to remove to get the action name.
-  // Type name: Fibonacci_FeedbackMessage -> Action name: Fibonacci
-  const std::string action_name = is_action_type ? remove_action_subtype(type_name) : "";
-  const std::string filename = is_action_type
-                                 ? action_name + ".action"
-                                 : type_name + extension_for_format(definition_identifier.format);
+  std::string filename = "";
+  if (subfolder == "action") {
+    // The action type name includes the subtype which we have to remove to get the action name.
+    // Type name: Fibonacci_FeedbackMessage -> Action name: Fibonacci
+    filename = remove_action_subtype(type_name) + ".action";
+  } else if (subfolder == "srv") {
+    // The service type name includes the subtype which we have to remove to get the service name.
+    // Type name: SetBool_Request -> Service name: SetBool
```

Comment:
I thought these don't exist anymore? Does the code have to handle both with & without the suffix because they exist in <=iron but not in >iron?

Reply:
We still use `_Request` and `_Response` as type name suffixes when advertising the services as we can't (or rather shouldn't) advertise request and response types with the same type name.